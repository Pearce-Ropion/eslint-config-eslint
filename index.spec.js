const eslintConfig = require('./index');

describe('ESLint Configuration', () => {

    it('exports eslint config', () => {
        expect(eslintConfig).toMatchSnapshot();
    });

});
