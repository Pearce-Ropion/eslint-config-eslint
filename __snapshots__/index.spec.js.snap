// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ESLint Configuration exports eslint config 1`] = `
Object {
  "env": Object {
    "es6": true,
    "node": true,
  },
  "parserOptions": Object {
    "ecmaVersion": 2018,
  },
  "rules": Object {
    "array-bracket-newline": Array [
      "error",
    ],
    "arrow-body-style": Array [
      "error",
      "as-needed",
      Object {
        "requireReturnForObjectLiteral": true,
      },
    ],
    "arrow-parens": Array [
      "error",
      "as-needed",
    ],
    "arrow-spacing": Array [
      "error",
    ],
    "block-scoped-var": Array [
      "error",
    ],
    "block-spacing": Array [
      "error",
    ],
    "brace-style": Array [
      "error",
    ],
    "camelcase": Array [
      "error",
    ],
    "comma-dangle": Array [
      "warn",
      "always-multiline",
    ],
    "comma-spacing": Array [
      "error",
    ],
    "comma-style": Array [
      "error",
    ],
    "complexity": Array [
      "error",
    ],
    "computed-property-spacing": Array [
      "error",
      "never",
    ],
    "consistent-this": Array [
      "error",
    ],
    "constructor-super": Array [
      "error",
    ],
    "curly": Array [
      "error",
    ],
    "default-case": Array [
      "error",
    ],
    "dot-location": Array [
      "error",
      "property",
    ],
    "dot-notation": Array [
      "error",
    ],
    "eol-last": Array [
      "error",
    ],
    "eqeqeq": Array [
      "error",
    ],
    "for-direction": Array [
      "error",
    ],
    "func-call-spacing": Array [
      "error",
    ],
    "generator-star-spacing": Array [
      "error",
    ],
    "getter-return": Array [
      "error",
    ],
    "global-require": Array [
      "error",
    ],
    "guard-for-in": Array [
      "error",
    ],
    "indent": Array [
      "error",
      4,
      Object {
        "SwitchCase": 1,
      },
    ],
    "key-spacing": Array [
      "error",
    ],
    "keyword-spacing": Array [
      "error",
    ],
    "linebreak-style": Array [
      "error",
    ],
    "lines-around-comment": Array [
      "error",
      Object {
        "beforeBlockComment": true,
      },
    ],
    "lines-between-class-members": Array [
      "error",
      "always",
    ],
    "max-depth": Array [
      "warn",
      6,
    ],
    "max-nested-callbacks": Array [
      "error",
      5,
    ],
    "max-params": Array [
      "warn",
      6,
    ],
    "max-statements-per-line": Array [
      "error",
    ],
    "new-parens": Array [
      "error",
    ],
    "no-alert": Array [
      "error",
    ],
    "no-array-constructor": Array [
      "error",
    ],
    "no-await-in-loop": Array [
      "error",
    ],
    "no-bitwise": Array [
      "warn",
    ],
    "no-caller": Array [
      "error",
    ],
    "no-case-declarations": Array [
      "error",
    ],
    "no-catch-shadow": Array [
      "error",
    ],
    "no-class-assign": Array [
      "error",
    ],
    "no-compare-neg-zero": Array [
      "error",
    ],
    "no-cond-assign": Array [
      "error",
    ],
    "no-const-assign": Array [
      "error",
    ],
    "no-control-regex": Array [
      "error",
    ],
    "no-delete-var": Array [
      "error",
    ],
    "no-div-regex": Array [
      "error",
    ],
    "no-dupe-args": Array [
      "error",
    ],
    "no-dupe-class-members": Array [
      "error",
    ],
    "no-dupe-keys": Array [
      "error",
    ],
    "no-duplicate-case": Array [
      "error",
    ],
    "no-else-return": Array [
      "error",
    ],
    "no-empty": Array [
      "error",
    ],
    "no-empty-character-class": Array [
      "error",
    ],
    "no-empty-function": Array [
      "error",
    ],
    "no-empty-pattern": Array [
      "error",
    ],
    "no-eq-null": Array [
      "error",
    ],
    "no-eval": Array [
      "error",
    ],
    "no-ex-assign": Array [
      "error",
    ],
    "no-extend-native": Array [
      "error",
    ],
    "no-extra-bind": Array [
      "error",
    ],
    "no-extra-boolean-cast": Array [
      "error",
    ],
    "no-extra-label": Array [
      "error",
    ],
    "no-extra-parens": Array [
      "error",
    ],
    "no-fallthrough": Array [
      "error",
    ],
    "no-floating-decimal": Array [
      "error",
    ],
    "no-func-assign": Array [
      "error",
    ],
    "no-global-assign": Array [
      "error",
    ],
    "no-implicit-coercion": Array [
      "error",
    ],
    "no-implicit-globals": Array [
      "error",
    ],
    "no-implied-eval": Array [
      "error",
    ],
    "no-inner-declarations": Array [
      "error",
    ],
    "no-invalid-regexp": Array [
      "error",
    ],
    "no-invalid-this": Array [
      "error",
    ],
    "no-irregular-whitespace": Array [
      "error",
    ],
    "no-iterator": Array [
      "error",
    ],
    "no-label-var": Array [
      "error",
    ],
    "no-labels": Array [
      "error",
    ],
    "no-lone-blocks": Array [
      "error",
    ],
    "no-lonely-if": Array [
      "error",
    ],
    "no-loop-func": Array [
      "error",
    ],
    "no-mixed-requires": Array [
      "error",
      Object {
        "grouping": true,
      },
    ],
    "no-mixed-spaces-and-tabs": Array [
      "error",
    ],
    "no-multi-assign": Array [
      "error",
    ],
    "no-multi-spaces": Array [
      "error",
    ],
    "no-multi-str": Array [
      "error",
    ],
    "no-nested-ternary": Array [
      "error",
    ],
    "no-new": Array [
      "error",
    ],
    "no-new-func": Array [
      "error",
    ],
    "no-new-object": Array [
      "error",
    ],
    "no-new-require": Array [
      "error",
    ],
    "no-new-symbol": Array [
      "error",
    ],
    "no-new-wrappers": Array [
      "error",
    ],
    "no-octal": Array [
      "error",
    ],
    "no-octal-escape": Array [
      "error",
    ],
    "no-param-reassign": Array [
      "error",
    ],
    "no-path-concat": Array [
      "error",
    ],
    "no-proto": Array [
      "error",
    ],
    "no-redeclare": Array [
      "error",
    ],
    "no-restricted-imports": Array [
      "error",
    ],
    "no-restricted-properties": Array [
      "error",
    ],
    "no-return-assign": Array [
      "error",
    ],
    "no-return-await": Array [
      "error",
    ],
    "no-script-url": Array [
      "error",
    ],
    "no-self-assign": Array [
      "error",
    ],
    "no-self-compare": Array [
      "error",
    ],
    "no-sequences": Array [
      "error",
    ],
    "no-shadow": Array [
      "warn",
    ],
    "no-sparse-arrays": Array [
      "error",
    ],
    "no-sync": Array [
      "warn",
    ],
    "no-tabs": Array [
      "error",
    ],
    "no-template-curly-in-string": Array [
      "error",
    ],
    "no-this-before-super": Array [
      "error",
    ],
    "no-throw-literal": Array [
      "error",
    ],
    "no-trailing-spaces": Array [
      "error",
    ],
    "no-undef": Array [
      "error",
    ],
    "no-undef-init": Array [
      "error",
    ],
    "no-undefined": Array [
      "error",
    ],
    "no-unexpected-multiline": Array [
      "error",
    ],
    "no-unmodified-loop-condition": Array [
      "error",
    ],
    "no-unneeded-ternary": Array [
      "error",
    ],
    "no-unreachable": Array [
      "error",
    ],
    "no-unsafe-finally": Array [
      "error",
    ],
    "no-unsafe-negation": Array [
      "error",
    ],
    "no-unused-expressions": Array [
      "error",
    ],
    "no-unused-labels": Array [
      "error",
    ],
    "no-unused-vars": Array [
      "error",
    ],
    "no-use-before-define": Array [
      "error",
    ],
    "no-useless-call": Array [
      "error",
    ],
    "no-useless-computed-key": Array [
      "error",
    ],
    "no-useless-concat": Array [
      "error",
    ],
    "no-useless-constructor": "off",
    "no-useless-escape": Array [
      "error",
    ],
    "no-useless-return": Array [
      "error",
    ],
    "no-var": Array [
      "error",
    ],
    "no-void": Array [
      "error",
    ],
    "no-warning-comments": "off",
    "no-whitespace-before-property": Array [
      "error",
    ],
    "no-with": Array [
      "error",
    ],
    "object-shorthand": Array [
      "error",
    ],
    "prefer-const": Array [
      "error",
    ],
    "prefer-destructuring": "off",
    "prefer-numeric-literals": Array [
      "error",
    ],
    "prefer-promise-reject-errors": "off",
    "prefer-rest-params": Array [
      "error",
    ],
    "prefer-spread": Array [
      "error",
    ],
    "prefer-template": Array [
      "error",
    ],
    "radix": Array [
      "error",
    ],
    "require-await": Array [
      "error",
    ],
    "require-jsdoc": "off",
    "require-yield": "off",
    "rest-spread-spacing": Array [
      "error",
      "never",
    ],
    "semi": Array [
      "error",
    ],
    "semi-style": Array [
      "error",
      "last",
    ],
    "space-before-blocks": Array [
      "error",
    ],
    "space-before-function-paren": "off",
    "space-infix-ops": Array [
      "error",
    ],
    "switch-colon-spacing": Array [
      "error",
    ],
    "symbol-description": Array [
      "error",
    ],
    "template-curly-spacing": Array [
      "error",
    ],
    "template-tag-spacing": Array [
      "error",
    ],
    "use-isnan": Array [
      "error",
    ],
    "valid-jsdoc": "off",
    "valid-typeof": Array [
      "error",
    ],
    "vars-on-top": Array [
      "error",
    ],
    "wrap-iife": Array [
      "error",
    ],
    "yield-star-spacing": Array [
      "error",
    ],
    "yoda": Array [
      "error",
    ],
  },
}
`;
